models/job.go

```diff
@@ -361,6 +361,8 @@ type JobService interface {
 	Deploy(context.Context, string, string, []JobSpec, progress.Observer) (DeploymentID, error)
 	// GetDeployment getting status and result of job deployment
 	GetDeployment(ctx context.Context, deployID DeploymentID) (JobDeployment, error)
+	// GetWithFilts gets the jobspec based on projectName, jobName, resourceDestination filters.
+	GetWithFilters(ctx context.Context, projectName string, jobName string, resourceDestination string) ([]JobSpec, error)
```

Comment:
Please change it to:

```go
type JobService interface {
    ...
    GetByFilter(ctx context.Context, filter JobSpecFilter) ([]JobSpec, error)
    ...
}
```

and for the `JobSpecFilter`:

```go
type JobSpecFilter struct {
    ProjectName               string
    JobName                     string
    ResourceDestination  string
}
```

With this, if there's any change to the filter, we don't need to update the interface. cc: @arinda-arif